homeassistant/components/modbus/climate.py

```diff
@@ -90,6 +96,30 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     temp_step = config[CONF_STEP]
     hub_name = config[CONF_HUB]
     hub = hass.data[MODBUS_DOMAIN][hub_name]
+    structure = config[CONF_STRUCTURE]
+
+    if data_type != DATA_TYPE_CUSTOM:
+        try:
+            structure = f">{DEFAULT_STRUCT_FORMAT[data_type][count]}"
+        except KeyError:
+            _LOGGER.error(
+                "Climate %s: Unable to find a data type matching count value %s, try a custom type",
+                name,
+                count,
+            )
+            return
+
+    try:
+        size = struct.calcsize(structure)
+    except struct.error as err:
+        _LOGGER.error("Error in sensor %s structure: %s", name, err)
+        return
+
+    if count * 2 != size:
+        _LOGGER.error(
+            "Structure size (%d bytes) mismatch registers count (%d words)", size, count
+        )
+        return
```

Comment:
We have test cases for sensor, could you please add a couple of cases testing the structure. Remark the reason for the test cases is not to test your code, but to secure a future change does not break it.

Thanks in advance, I think the new much simpler way of adding test cases should make it easy to do, but just call if you need help.

Reply:
I will try to add some tests in the upcoming days. And sure, I will contact you if I need some help.

My experience with async tests in Python wasn't good so far. I don't know yet how to debug them. But maybe it's time to learn something new :)

Reply:
I use Visual Studio Code, when you open the workspace it tries “discover tests” when that runs successful you can run debug using the test entry.

Reply:
Thanks for the tip. I use PyCharm, but I'm also familiar with VS Code. In recent weeks I use it every day for my Arduino projects in PlatformIO. I'll try your solution.

Reply:
I used to use pycharm but switched to VC. Dif you add an extension to handle Arduino (I am no friend of the standard editor)?

Reply:
Yes, the extension is called [PlatformIO IDE](https://platformio.org/). They support many frameworks and boards, Arduino is just one of them. However, most of my projects at the moment are for Arduino.

Reply:
@janiversen I guess the only thing left in this PR is to write some unit tests for the Climate. I'll try to write some during this week. If I get stuck, I'll definitely ask for your help :)

Reply:
If @MartinHjelmare is up to it, then let us get this PR merged! For sure I would like test cases but seeing this code merged is bigger priority.

@MartinHjelmare gave me a good idea on how to simplify the test cases quite a lot by using a pytest decorator. I am preparing a PR that updates all test cases.

Reply:
OK, agree. While you'll work on test improvements, I will move to the other PR to get the work moving there too.